hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -208,6 +209,7 @@ public void sync() throws Exception {
           throw ex;
         } finally {
           deltaSyncService.ifPresent(DeltaSyncService::close);
+          Metrics.shutdown();
```

Comment:
this only works with sync once mode. can we also hook it up with continuous-mode's shutdown callback as well?

Reply:
The Metric's reporter already has a shutdown hook of its own, that should run when jvm exits. That should suffice the continuous mode case right ?

Reply:
Please correct me if I'm wrong. It seems like the DeltaStreamer continuous mode does not have this issue.

Reply:
functionality-wise, it should be ok. i think it's a question of where is the single place we can clean up all resources. `org.apache.hudi.utilities.deltastreamer.DeltaSync#close` looks like the right place. It looks more of a minor clean up to me now.